Common/include/toolboxes/C2DContainer.hpp

```diff
@@ -595,3 +595,33 @@ struct C3DDummyMiddleView
     return data(i,k);
   }
 };
+
+/* Decorate a vector type (Storage) with 3 dimensions */
+template<class Storage>
+struct C3DContainerDecorator {
+  using Scalar = typename Storage::Scalar;
+  using Index = typename Storage::Index;
+
+  Storage storage;
+  Index M, N;
+
+  C3DContainerDecorator() = default;
+
+  C3DContainerDecorator(Index length, Index rows, Index cols, Scalar value = 0) {
+    resize(length, rows, cols, value);
+  }
+
+  void resize(Index length, Index rows, Index cols, Scalar value = 0) {
+    M = rows;
+    N = cols;
+    storage.resize(length*rows*cols) = value;
+  }
+
+  Scalar& operator() (Index i, Index j, Index k) { return storage(i*M*N + j*N + k); }
+  const Scalar& operator() (Index i, Index j, Index k) const { return storage(i*M*N + j*N + k); }
+};
+
+/* Define an alias for a 3D int matrix, we use su2vector to store the integers contiguously
+ * and the container decorator to create the access semantics we want. */
+using C3DIntMatrix = C3DContainerDecorator<su2vector<int> >;
+using C3DDoubleMatrix = C3DContainerDecorator<su2vector<double> >;
```

Comment:
Always `su2double` unless you absolutely need a passive type, in those cases use `passivedouble` which is defined centrally.

Reply:
do you i should have C3DContainerDecorator<su2vector<**su2**double> instead?

Reply:
yup